generator/operation_test.go

```diff
@@ -27,35 +27,11 @@ import (
 func TestUniqueOperationNames(t *testing.T) {
 	doc, err := spec.Load("../fixtures/codegen/todolist.simple.yml")
 	if assert.NoError(t, err) {
-		sp := doc.Spec()
-		sp.Paths.Paths["/tasks"].Post.ID = "saveTask"
-		sp.Paths.Paths["/tasks"].Post.AddExtension("origName", "createTask")
-		sp.Paths.Paths["/tasks/{id}"].Put.ID = "saveTask"
-		sp.Paths.Paths["/tasks/{id}"].Put.AddExtension("origName", "updateTask")
-
-		ops := gatherOperations(doc, nil)
-		assert.Len(t, ops, 4)
-		_, exists := ops["saveTask"]
-		assert.True(t, exists)
-		_, exists = ops["saveTask1"]
-		assert.True(t, exists)
-	}
-}
-
-func TestEmptyOperationNames(t *testing.T) {
-	doc, err := spec.Load("../fixtures/codegen/todolist.simple.yml")
-	if assert.NoError(t, err) {
-		sp := doc.Spec()
-		sp.Paths.Paths["/tasks"].Post.ID = ""
-		sp.Paths.Paths["/tasks"].Post.AddExtension("origName", "createTask")
-		sp.Paths.Paths["/tasks/{id}"].Put.ID = ""
-		sp.Paths.Paths["/tasks/{id}"].Put.AddExtension("origName", "updateTask")
-
```

Comment:
TestUniqueOperationNames still runs, but since the name no longer relies on Post.ID, I removed them.  The exist assertions checks for uniqueness in the names for the path /tasks with different methods.

TestEmptyOperationNames didn't seem valid anymore since the new naming convention doesn't rely on Post.ID (which can be empty) anymore.  Paths are required, and method is also (...hopefully).